### PR TITLE
Switch record_payloads to bytes::Buf

### LINKED ISSUE #29: Switch record_payloads from using &'b [u8] to bytes::Buf
As we discussed in  quickwit-oss/quickwit#2834, we need to store more than a record's content in payload. In case we want to store a command like this:
```
enum Command<'a> {
     Ingest {
         json_payload: &'a [u8]
     },
     Commit,
     // ... more to come?
}
```
we will need to write a command type followed by `json_payload`. Having `&[u8]` in the interface doesn't allow us to write two chunks of data efficiently. However, if we switch interface to use [`bytes::Buf`](https://docs.rs/bytes/latest/bytes/buf/trait.Buf.html) it will allows us to continue using `&[u8]` thanks to [`impl Buf for &[u8]`](https://docs.rs/bytes/latest/bytes/buf/trait.Buf.html#impl-Buf-for-%26%5Bu8%5D) and at the same time be able to provide access to command code and payload without copying thanks to Buf's support for [chaining](https://docs.rs/bytes/latest/bytes/buf/trait.Buf.html#method.chain).

### PR DESCRIPTION
**Pull request by imotov**
Switches record_payloads internface from using &'b [u8] to bytes::Buf.

Closes #29

### CODE REVIEW

**Comment by fulmicoton** [comment]
@trinity-1686a can you review?

File: src/record.rs [hunk 1]
```diff
@@ -141,25 +143,30 @@ impl<'a> MultiRecord<'a> {
...
-            assert!(record_payload.len() <= u32::MAX as usize);
+        for (position, mut record_payload) in record_payloads {
+            assert!(record_payload.remaining() <= u32::MAX as usize);
             // TODO add assert for position monotonicity?
+            let record_payload = &mut record_payload;
             output.extend_from_slice(&position.to_le_bytes());
-            output.extend_from_slice(&(record_payload.len() as u32).to_le_bytes());
-            output.extend_from_slice(record_payload);
+            output.extend_from_slice(&(record_payload.remaining() as u32).to_le_bytes());
+            while record_payload.remaining() > 0 {
```
Comment (trinity-1686a) [inline src/record.rs:165]: nit:
```suggestion
            while record_payload.has_remaining() {
```

**Review by trinity-1686a — APPROVED** [review]
LGTM. Just one small nit

**Comment by trinity-1686a** [comment]
thanks!

**Pull request merged**